aten/src/ATen/native/mkldnn/BinaryOps.cpp

```diff
@@ -46,6 +47,35 @@ Tensor& mkldnn_mul_(Tensor& self, const Tensor& other) {
 namespace at {
 namespace native {
 
+Tensor emptyBinaryOp(const Tensor& self, const Tensor& other) {
+  if (!self.requires_grad() && !other.requires_grad()) {
+    auto out_size = infer_size(self.sizes(), other.sizes());
```

Comment:
what if shapes are incompatible? Does `infer_size` throw?

Reply:
yea, i'm 99% sure I checked and it does. i'll add a test

Reply:
this test was added